packages/mdast-util-split/src/index.js

```diff
@@ -0,0 +1,124 @@
+module.exports = split
+import visit from 'unist-util-visit'
+
+function split (tree, {splitDepth = 1,
+  introductionAsProperty = true, conclusionAsProperty = false}) {
+  if (splitDepth < 1) {
+    throw new Error('splitDepth must be greater than 1')
+  }
+  const firstHeading = find(tree, {type: 'heading', depth: 1})
+  if (!firstHeading) {
+    throw new Error('No heading')
+  }
+  const splitter = new Splitter()
+  visit(tree, null, (node, index, parent) => splitter.visit(node, index, parent))
+  splitter.introduction = splitter.treeBefore.children
+  if (conclusionAsProperty) {
+    splitter.extractConclusions()
+  }
+  if (introductionAsProperty) {
+    splitter.extractIntroductions()
+  }
+  return {
+    introduction: splitter.introduction,
+    trees: splitter.subTrees,
+  }
+}
+
+function find (tree, {type = 'heading', depth = 1}) {
+  for (let i = 0; i < tree.children.length; ++i) {
+    if (tree.children[i].type === type && tree.children[i].depth === depth) {
+      return tree.children[i]
+    }
+  }
+  return null
+}
+
+class Splitter {
+  constructor (depth = 1) {
+    this.lastIndex = -1
+    this.treeBefore = {
```

Comment:
What is `treeBefore`?

Reply:
Removed, was something I created in a first attempt to extract introduction.